mobile-app/app/hooks/useIsBlockchainDown.ts

```diff
@@ -0,0 +1,34 @@
+import { useEffect, useState } from 'react'
+import { useSelector } from 'react-redux'
+import { RootState } from '@store'
+
+const MAX_TIME_DIFF = 45 * 60 * 1000 // 45 mins
+
+export function useIsBlockchainDown (): boolean {
+    const blockLastSync = useSelector((state: RootState) => state.block.lastSync)
+    const [isBlockchainDown, setIsBlockchainDown] = useState(false)
+
+    useEffect(() => {
+        function isBlockchainDownFn (): boolean {
+            const nowEpoch = Date.now()
+            let blockLastSyncTime: number | undefined
+
+            if (blockLastSync !== undefined && blockLastSyncTime !== undefined) {
+                blockLastSyncTime = Date.parse(blockLastSync) // convert date string to date number
+                const timeDifference = nowEpoch - blockLastSyncTime
```

Comment:
I believe we are using this library. Maybe you can utilize the functions here to get a more accurate result of dates - https://day.js.org/docs/en/installation/installation